File: frontend/src/app/services/auth.service.ts
// src/app/services/auth.service.ts

import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

@Injectable({ providedIn: 'root' })
export class AuthService {
  private API_URL = 'http://localhost:8000/api';

  constructor(private http: HttpClient) {}

  // Registro de usuario
  register(username: string, password: string): Observable<any> {
    return this.http.post(`${this.API_URL}/register/`, {
      username,
      password,
    });
  }

  // Login (JWT)
  login(username: string, password: string): Observable<{ access: string; refresh: string }> {
    return this.http.post<{ access: string; refresh: string }>(
      `${this.API_URL}/login/`,
      { username, password }
    );
  }

  // Obtener perfil del usuario autenticado
  getProfile(token: string): Observable<any> {
    const headers = new HttpHeaders().set('Authorization', `Bearer ${token}`);
    return this.http.get(`${this.API_URL}/me/`, { headers });
  }

  // Guardar token en localStorage
  saveToken(token: string): void {
    localStorage.setItem('access_token', token);
  }

  // Obtener token
  getToken(): string | null {
    return localStorage.getItem('access_token');
  }

  // Cerrar sesión
  logout(): void {
    localStorage.removeItem('access_token');
  }

  // Verificar si está logueado
  isLoggedIn(): boolean {
    return this.getToken() !== null;
  }
}

File: registros/lib/service/auth.service.dart
import 'dart:convert';
import 'package:http/http.dart' as http;

class AuthService {
  static const String baseUrl = 'http://localhost:8000/api';

  static Future<http.Response> login(String username, String password) async {
    final url = Uri.parse('$baseUrl/login/');
    final response = await http.post(
      url,
      headers: {'Content-Type': 'application/json'},
      body: jsonEncode({'username': username, 'password': password}),
    );
    return response;
  }

  static Future<http.Response> register(
    String username,
    String email,
    String password, {
    String? firstName,
    String? lastName,
  }) async {
    final url = Uri.parse('$baseUrl/register/');
    final response = await http.post(
      url,
      headers: {'Content-Type': 'application/json'},
      body: jsonEncode({
        'username': username,
        'email': email,
        'password': password,
        'first_name': firstName,
        'last_name': lastName,
      }),
    );
    return response;
  }

  static Future<http.Response> getProfile(String accessToken) async {
    final url = Uri.parse('$baseUrl/me/');
    final response = await http.get(
      url,
      headers: {'Authorization': 'Bearer $accessToken'},
    );
    return response;
  }
}
